src/addresses.rs

```diff
@@ -87,12 +88,22 @@ fn build_new_addr(
     poi: &Poi,
     admins: Vec<Arc<mimir::Admin>>,
 ) -> mimir::Address {
-    let postcode = poi
+    let postcodes = poi
         .properties
         .iter()
         .find(|p| &p.key == "addr:postcode")
-        .map(|p| p.value.to_owned());
-    let postcodes = postcode.map_or(vec![], |p| vec![p]);
+        .map_or_else(
+            || {
+                admins
+                    .iter()
+                    .filter(|admin| !admin.zip_codes.is_empty())
```

Comment:
I would keep only admins with a single zip_code. I am not sure that it would be useful to associate (and index) a POI with several postcodes inherited from the city, most of them being probably wrong for this specific place.
```suggestion
                    .filter(|admin| admin.zip_codes.len() == 1)
```